doc/Programming-guide/Key-platform-features/Endpoint-profiles/index.md

```diff
@@ -6,117 +6,397 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-profiles/
-The client-side endpoint profile is unidirectionally synchronized, and thus should not be considered as a means to temporarily store endpoint data in the Kaa cluster. There is no way for the endpoint to retrieve the profile information back from the Kaa cluster. At its start, the application must fill in the current endpoint profile with the up-to-date data and execute corresponding SDK API calls. The endpoint SDK does not persist the profile information over the endpoint reboots. However, it detects profile data changes and submits the new data to the Kaa cluster as a profile update.
-
 
-![ClientSideEndpointProfileGeneration](ClientSideEndpointProfileGeneration_0_8_0.png "ClientSideEndpointProfileGeneration")
+The client-side endpoint profile is initially generated at the stage of a new endpoint registration. Prior to that, the client-side endpoint profile values should be 
+specified by the client developer using the Kaa endpoint SDK. The client-side endpoint profile can be also updated at run time using SDK API call. In this case, SDK executes 
+profile update request and endpoint membership in the endpoint groups is re-evaluated and updated to match the new endpoint profile.
+The client-side endpoint profile is unidirectionally synchronized, and thus should not be considered as a means to temporarily store endpoint data in the Kaa cluster. 
+There is no way for the endpoint to retrieve the profile information back from the Kaa cluster. At its start, the application must fill in the current endpoint profile 
+with the up-to-date data and execute corresponding SDK API calls. The endpoint SDK does not persist the profile information over the endpoint reboots. However, 
+it detects profile data changes and submits the new data to the Kaa cluster as a profile update.
 
+<img src="endpoint-profile-generation/ClientSideEndpointProfileGeneration_0_8_0.png">
 
-## Client-side endpoint profile update ##
-When the endpoint profile information changes as the result of the client operation or user's actions, it is the client implementation responsibility to update the profile via the endpoint SDK API calls. The endpoint SDK detects profile changes by comparing the new profile hash against the previously persisted one. Should there be a change, the endpoint profile management module passes it to the Operations server, which in turn updates the endpoint profile information in the database and revises the endpoint groups membership.
-Client-side endpoint profile example
-The following code block provides a simple client-side endpoint profile schema example.
+### Client-side endpoint profile example ###
 
+For the purpose of this guide we will use a fairly abstract client-side profile schema shown below.
```

Comment:
Could the schema be described a bit?
E.g. "The schema defines 4 fields: ...". This will let a user some notion regarding what the schema is for. Especially if the user is not familiar with the related technologies.